src/test/allocator_tests.cpp

```diff
@@ -15,11 +15,9 @@ BOOST_FIXTURE_TEST_SUITE(allocator_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(arena_tests)
 {
-    // Fake memory base address for testing
-    // without actually using memory.
-    void *synth_base = reinterpret_cast<void*>(0x08000000);
     const size_t synth_size = 1024*1024;
-    Arena b(synth_base, synth_size, 16);
+    std::unique_ptr<char[]> synth_base(new char[synth_size]());
+    Arena b(synth_base.get(), synth_size, 16);
```

Comment:
Using a fake address here was on purpose, it would catch accesses to the underlying memory by causing a segfault.

Reply:
I'm slightly confused. I don't think `walk` is supposed to access the contents of the memory, it is supposed to be a debugging function that prints the allocation metadata, addresses and such, which is intentionally stored outside of the arena (to save on scarce page-locked memory). If it accesses, or even dereferences, the underlying memory that seems like a bug to me.

Reply:
I'm not sure why, but Inserting `base` into `std::cout` inside `printchunk` causes the segfault when executed with `make CPPFLAGS=-DARENA_DEBUG check`:

```console
Entering test module "Bitcoin Core Test Suite"
test/allocator_tests.cpp(14): Entering test suite "allocator_tests"
test/allocator_tests.cpp(16): Entering test case "arena_tests"
0xunknown location(0): fatal error: in "allocator_tests/arena_tests": memory access violation at address: 0x080ffc10: no mapping at fault address                                        
test/allocator_tests.cpp(16): last checkpoint: "arena_tests" entry.
Test is aborted
test/allocator_tests.cpp(16): Leaving test case "arena_tests"; testing time: 8196us
Test is aborted
test/allocator_tests.cpp(14): Leaving test suite "allocator_tests"; testing time: 8231us
Test is aborted
```

The segfault does not occur if I remove `base` from being inserted into the stream.

Reply:
Turns out this segfault can be avoided instead by changing `printchunk` to take `void*` rather than `char*`, which is what I had been doing in #16195. Updated in 30fb598.